### PR TITLE
PT-2299 - collect openssl x509 certificate information for each secret

### PR DESCRIPTION
**Pull request by svetasmirnova**
- Implemented the feature
- Added test case

- [x] The contributed code is licensed under GPL v2.0
- [x] Contributor Licence Agreement (CLA) is signed
- [ ] util/update-modules has been ran
- [ ] Documentation updated
- [x] Test suite update

### CODE REVIEW

File: src/go/pt-k8s-debug-collector/dumper/dumper.go [hunk 1]
```diff
@@ -6,6 +6,7 @@ import (
 	"compress/gzip"
 	"encoding/base64"
 	"encoding/json"
+	"html/template"
```
Comment (artemgavrilov) [inline src/go/pt-k8s-debug-collector/dumper/dumper.go]: Most likely you want here `text/template`

**Review by artemgavrilov — APPROVED** [review]
LGTM with comment

File: src/go/pt-k8s-debug-collector/dumper/dumper.go [hunk 2]
```diff
@@ -113,8 +122,36 @@ func New(location, namespace, resource string, kubeconfig string, forwardport st
 			"perconaservermongodbs",
 		)
 	}
+	sslSecrets := make([]sslSecret, 0)
 	filePaths := make([]string, 0)
-	if resourceType(resource) == "pxc" {
+	switch resource {
```
Comment (ihanick) [inline src/go/pt-k8s-debug-collector/dumper/dumper.go]: it should be a`switch resourceType(resource)`.
with plain `switch resource`, it's not working (and breaks pxc) the new code breaks
`pt-k8s-debug-collector -cluster cluster1 -namespace pgo -resource pgv2`
the dumper receives pgv2/cluster1 string inside resource variable.

**Review by ihanick — CHANGES_REQUESTED** [review]
not working if -cluster argument is specified, instead of switch resource it should be switch resourceType(resource)

File: src/go/pt-k8s-debug-collector/dumper/dumper.go (hunk 1 above)
Comment (ihanick) [reply src/go/pt-k8s-debug-collector/dumper/dumper.go]: I've tried text/template and it works as expected

File: src/go/pt-k8s-debug-collector/dumper/dumper.go [hunk 3]
```diff
@@ -126,8 +163,32 @@ func New(location, namespace, resource string, kubeconfig string, forwardport st
 			"var/lib/mysql/auto.cnf",
 		)
 		d.fileContainer = "logs"
+		sslSecrets = append(sslSecrets,
+			sslSecret{
+				secret:    "{{ .Name }}-ssl",
+				resource:  "pxc",
+				dataNames: []string{"ca.crt", "tls.crt"},
+			},
+		)
```
Comment (ihanick) [inline src/go/pt-k8s-debug-collector/dumper/dumper.go:192]: another two required for v1
```
+                       sslSecret{
+                               secret:    "{{ .Name }}-ssl-keypair",
+                               resource:  "perconapgclusters",
+                               dataNames: []string{"tls.crt"},
+                       },
+                       sslSecret{
+                               secret:    "{{ .Name }}-replication-ssl-keypair",
+                               resource:  "perconapgclusters",
+                               dataNames: []string{"tls.crt"},
+                       },
```

File: src/go/pt-k8s-debug-collector/dumper/dumper.go [hunk 4]
```diff
@@ -113,8 +122,36 @@ func New(location, namespace, resource string, kubeconfig string, forwardport st
...
 	}
+	sslSecrets := make([]sslSecret, 0)
 	filePaths := make([]string, 0)
-	if resourceType(resource) == "pxc" {
+	switch resourceType(resource) {
+	case "pg":
+		sslSecrets = append(sslSecrets,
+			sslSecret{
+				secret:    "{{ .Name }}-ssl-ca",
+				resource:  "perconapgclusters",
```
Comment (ihanick) [inline src/go/pt-k8s-debug-collector/dumper/dumper.go]: here and in https://github.com/percona/percona-toolkit/blob/9b3d92997a32395c7082b5990e363275c60c1711/src/go/pt-k8s-debug-collector/dumper/dumper.go#L85 it's better to use a resource type name with percona's domain. If another operator with same short name is installed (or v1 and v2 ) it causes conflicts
```
-                       "perconapgclusters",
-                       "pgclusters",
-                       "pgpolicies",
-                       "pgreplicas",
-                       "pgtasks",
+                       "perconapgclusters.pg.percona.com",
+                       "pgclusters.pg.percona.com",
+                       "pgpolicies.pg.percona.com",
+                       "pgreplicas.pg.percona.com",
+                       "pgtasks.pg.percona.com",
                )
        case "pgv2":
                resources = append(resources,
-                       "perconapgbackups",
-                       "perconapgclusters",
-                       "perconapgrestores",
+                       "perconapgbackups.pgv2.percona.com",
+                       "perconapgclusters.pgv2.percona.com",
+                       "perconapgrestores.pgv2.percona.com",
```

**Review by ihanick — CHANGES_REQUESTED** [review]
please fix comments related to text/template, add missing certificates for pg v1, potentially update the resource name to the full one. for v1, the secret name could be changed

**Comment by svetasmirnova** [comment]
> not working if -cluster argument is specified, instead of switch resource it should be switch resourceType(resource)

This is a known issue: https://perconadev.atlassian.net/browse/PT-2189

File: src/go/pt-k8s-debug-collector/dumper/dumper.go [hunk 5]
```diff
@@ -82,39 +91,77 @@ func New(location, namespace, resource string, kubeconfig string, forwardport st
...
+			"pgpolicies.pg.percona.com",
+			"pgreplicas.pg.percona.com",
+			"pgtasks.pg.percona.com",
 		)
 	case "pgv2":
 		resources = append(resources,
-			"perconapgbackups",
-			"perconapgclusters",
-			"perconapgrestores",
+			".perconapgbackups.pgv2.percona.com",
```
Comment (svetasmirnova) [inline src/go/pt-k8s-debug-collector/dumper/dumper.go]:
```suggestion
			"perconapgbackups.pgv2.percona.com",
```

**Review by ihanick — APPROVED** [review]
LGTM

**Pull request merged**